toplevel/coqtop.ml

```diff
@@ -665,52 +753,62 @@ let init_toplevel arglist =
         exit 1
       end;
       Flags.if_verbose print_header ();
-      inputstate ();
       Mltop.init_known_plugins ();
       engage ();
+
+      (* This state will be shared by all the documents *)
+      Stm.init_core ();
```

Comment:
I'm uneasy with the name of this function. I think the name should reflect that it snapshots the father of all initial states, as your comment says. What about `Stm.define_initial_doc_state ()`

Reply:
You have a point, but umm, it maybe will do more than defining the initial state; I could well see it starting the worker manager etc...